algorithm/eight_queens.py

```diff
@@ -29,15 +34,155 @@ def board(self) -> list[list[dict]]:
     def queens(self) -> list[tuple[int,int]]:
         return self._queen_positions
     
-    def add_queen(self, position: tuple[int,int]):
-        self._validate_position(position)
-        self._queen_positions.append(position)
-    
     def place_queen(self, position: tuple):
         self._validate_position(position)
+        self._queen_positions.append(position)
+
+        row, column = position
+        self._board[row][column]["id"] = "queen"
+
+        def tile_is_empty(row: int, column: int) -> bool:
+            "helper subfunction to avoid replacing a queen's id with a killzone from other queens"
+            return self._board[row][column]["id"] != "queen"
+
+        # row and column killzone
+        for i in range(self.size):
+            # rows
+            placed_queen = column
+            if i != placed_queen:
+                self._board[row][i]["value"] += 1
+
+                if tile_is_empty(row, i):
+                    self._board[row][i]["id"] = "killzone"
+                    self._board[row][i]["asset"] = self._killzone_asset
```

Comment:
this code snippet is repeated multiple times, might be better to create a sub function for it?
```suggestion
        def add_killzone(coord, i):
                if is_tile_empty(coord, i):
                    self._board[coord][i]["id"] = "killzone"
                    self._board[coord][i]["asset"] = self._killzone_asset
```